subprojects/version-control/src/main/java/org/gradle/vcs/internal/DefaultVersionControlSystemFactory.java

```diff
@@ -59,14 +65,43 @@ public VersionControlSystem create(VersionControlSpec spec) {
         } else {
             vcs = new GitVersionControlSystem();
         }
-        return new LockingVersionControlSystem(vcs, vcsWorkingDirCache);
+        return new LockingVersionControlSystem(
+            new RequireSettingsVersionControlSystem(vcs, buildLayoutFactory),
+            vcsWorkingDirCache);
     }
 
     @Override
     public void stop() {
         vcsWorkingDirCache.close();
     }
 
+    private static final class RequireSettingsVersionControlSystem implements VersionControlSystem {
+        private final VersionControlSystem delegate;
+        private final BuildLayoutFactory buildLayoutFactory;
+
+        private RequireSettingsVersionControlSystem(VersionControlSystem delegate, BuildLayoutFactory buildLayoutFactory) {
+            this.delegate = delegate;
+            this.buildLayoutFactory = buildLayoutFactory;
+        }
+
+        @Override
+        public Set<VersionRef> getAvailableVersions(VersionControlSpec spec) {
+            return delegate.getAvailableVersions(spec);
+        }
+
+        @Override
+        public File populate(File versionDir, VersionRef ref, VersionControlSpec spec) {
+            File workingDir = delegate.populate(versionDir, ref, spec);
+            PluginRequests pluginRequests = ((AbstractVersionControlSpec)spec).getPluginRequests();
+            if (!pluginRequests.isEmpty() && buildLayoutFactory.findExistingSettingsFileIn(workingDir) == null) {
+                // Injecting plugins and no settings file exists.
+                // TODO: For now, just touch an empty file so this appears to be a well-formed build.
+                // Eventually, we may want to relax this requirement for included builds
```

Comment:
Why not just do that instead of this workaround? I've never understood why we care.

Reply:
yeah, I think we can probably just do that.  I think we used to not set "do not search upwards".